src/main/kotlin/io/github/mojira/arisa/ArisaMain.kt

```diff
@@ -1,213 +1,32 @@
 package io.github.mojira.arisa
 
-import arrow.syntax.function.partially1
-import ch.qos.logback.classic.AsyncAppender
-import ch.qos.logback.classic.LoggerContext
-import com.github.napstr.logback.DiscordAppender
-import com.uchuhimo.konf.Config
-import com.uchuhimo.konf.source.yaml
-import io.github.mojira.arisa.domain.Issue
-import io.github.mojira.arisa.infrastructure.HelperMessageService
-import io.github.mojira.arisa.infrastructure.ProjectCache
-import io.github.mojira.arisa.infrastructure.config.Arisa
-import io.github.mojira.arisa.infrastructure.jira.connectToJira
-import io.github.mojira.arisa.infrastructure.jira.toDomain
-import io.github.mojira.arisa.registry.TicketQueryTimeframe
-import io.github.mojira.arisa.registry.getModuleRegistries
 import net.rcarz.jiraclient.JiraClient
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
-import java.io.File
-import java.lang.Long.max
-import java.time.Duration
-import java.time.Instant
-import java.time.temporal.ChronoUnit
 import java.util.concurrent.TimeUnit
 
 val log: Logger = LoggerFactory.getLogger("Arisa")
 
-private const val TIME_MINUTES = 5L
-private const val MAX_QUERY_TIMEFRAME_IN_MINUTES = 10L
-
-const val MAX_RESULTS = 50
-
-private const val RELOG_INTERVAL_IN_SECONDS = 5 * 60L
-private const val WAIT_TIME_AFTER_CONNECTION_ERROR_IN_SECONDS = 40L
-
+/**
+ * Global instance for the Jira client.
+ * (not nice but it works)
+ *
+ * Gets initialized by [JiraConnectionService.connect]
+ */
 lateinit var jiraClient: JiraClient
 
-@Suppress("LongMethod", "TooGenericExceptionCaught", "NestedBlockDepth")
-fun main() {
-    val config = readConfig()
-    setWebhookOfLogger(config)
-
-    jiraClient =
-        connectToJira(
-            config[Arisa.Credentials.username],
-            config[Arisa.Credentials.password],
-            config[Arisa.Issues.url]
-        )
-    log.info("Connected to jira")
-
-    // Get tickets for re-run and last run time
-    var lastRelog = Instant.now()
-    val lastRunFile = File("last-run")
-    val lastRun = readLastRun(lastRunFile)
-    var lastRunTime = readLastRunTime(lastRun)
-    var rerunTickets = lastRun.subList(1, lastRun.size).toSet()
-    val failedTickets = mutableSetOf<String>()
+val configService = ConfigService()
+val webhookService = WebhookService(configService.config)
+val connectionService = JiraConnectionService(configService.config)
+val executionService = ExecutionService(configService.config, connectionService)
```

Comment:
These could be `private`, or even better local variables of `main`; this would then most likely solve the build failure.